### PR TITLE
EJX magic for Ben

### PR DESCRIPTION
**Pull request by malomalo**

### CODE REVIEW

File: lib/ejx/template/subtemplate.rb [hunk 1]
```diff
@@ -6,69 +6,111 @@ def initialize(opening, modifiers, append: true)
     @children = [opening]
     @modifiers = modifiers
     @append = append
+    
+    @function = false
+    @arrow_function = false
+    @async = false
```
Comment (bemky) [inline lib/ejx/template/subtemplate.rb]:
```suggestion
    @is_function = false
    @is_arrow_function = false
    @is_async = false
```

File: lib/ejx/template/subtemplate.rb [hunk 2]
```diff
@@ -6,69 +6,111 @@ def initialize(opening, modifiers, append: true)
...
+          #{' '*(indentation)}var #{sub_global_output_var}_result = (\\1
+        JS
+      else
+        @children.first.sub(/((?:async)?(?:\s*\([^\)\(]*\)|(?:(?<=\()|\s+)[^\(\s]+)?\s*=>\s*\{\s*)\Z/m, <<~JS)
+          (...__args) => {
+          #{' '*(indentation)}var #{sub_global_output_var}_results = [];
+          #{' '*(indentation)}var #{sub_global_output_var}_promises = [];
+          #{' '*(indentation)}var #{sub_global_output_var}_result = (\\1
+        JS
+      end
```
Comment (bemky) [inline lib/ejx/template/subtemplate.rb]:
```suggestion
      output << @children.first.sub(@function ? function_regex : arrow_function_regex, <<~JS)
        (...__args) => {
        #{' '*(indentation)}var #{sub_global_output_var}_results = [];
        #{' '*(indentation)}var #{sub_global_output_var}_promises = [];
        #{' '*(indentation)}var #{sub_global_output_var}_result = (\\1
      JS
```

File: lib/ejx/template/subtemplate.rb [hunk 3]
```diff
@@ -6,69 +6,111 @@ def initialize(opening, modifiers, append: true)
...
     else
-      output << "#{' '*indentation}var #{global_output_var} = [];\n"
-      output << "#{' '*indentation}__ejx_append("
-      output << @children.first << "\n"
+      puts '!!!'
```
Comment (bemky) [inline lib/ejx/template/subtemplate.rb]:
```suggestion
```

**Review by bemky — APPROVED** [review]

**Comment by bemky** [comment]
Getting this error when plugging into rex
```
  
Condenser::SyntaxError (/assets/pdf/app/views/surveys/_show.js: Unexpected token, expected ")" (31:56)

  29 |   var __a_results = [];
  30 |   var __a_promises = [];
> 31 |   var __a_result = locals.survey.listings.map((...locals.__args) => {
     |                                                         ^
  32 |     var __b_results = [];
  33 |     var __b_promises = [];
  34 |     var __b_result = (async listing => {):
  
/assets/pdf/app/views/surveys/_show.js: Unexpected token, expected ")" (31:56)

  29 |   var __a_results = [];
  30 |   var __a_promises = [];
> 31 |   var __a_result = locals.survey.listings.map((...locals.__args) => {
     |                                                         ^
  32 |     var __b_results = [];
  33 |     var __b_promises = [];
  34 |     var __b_result = (async listing => {
/assets/pdf/app/views/surveys/_show.js: Unexpected token, expected ")" (31:56)

  29 |   var __a_results = [];
  30 |   var __a_promises = [];
> 31 |   var __a_result = locals.survey.listings.map((...locals.__args) => {
     |                                                         ^
  32 |     var __b_results = [];
  33 |     var __b_promises = [];
  34 |     var __b_result = (async listing => {
/assets/pdf/app/views/surveys/_show.js:31

```

**Comment by malomalo** [comment]
The Syntax Error should be solved with https://github.com/malomalo/condenser/commit/1ec65ab1ebd8f8785c176595fc89f5f77ba1c1cb

**Pull request merged**